state-chain/runtime/src/chainflip/solana_elections.rs

```diff
@@ -472,3 +510,103 @@ impl ElectionEgressWitnesser for SolanaEgressWitnessingTrigger {
 		})
 	}
 }
+
+#[derive(
+	Clone, PartialEq, Eq, Debug, Serialize, Deserialize, TypeInfo, Encode, Decode, PartialOrd, Ord,
+)]
+pub struct SolanaVaultSwapDetails {
+	pub from: SolAsset,
+	pub to: Asset,
+	pub deposit_amount: SolAmount,
+	pub destination_address: EncodedAddress,
+	pub deposit_metadata: Option<CcmDepositMetadata>,
+	// TODO: swap_account and creation_slot will be pulled into TransactionInId type (PRO-1760)
+	pub swap_account: SolAddress,
+	pub creation_slot: u64,
+	pub broker_fee: Beneficiary<AccountId>,
+	pub refund_params: ChannelRefundParameters,
+	pub dca_params: Option<DcaParameters>,
+	pub boost_fee: u8,
+	pub affiliate_fees: Affiliates<AffiliateShortId>,
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+impl BenchmarkValue for SolanaVaultSwapDetails {
+	fn benchmark_value() -> Self {
+		Self {
+			from: BenchmarkValue::benchmark_value(),
+			to: BenchmarkValue::benchmark_value(),
+			deposit_amount: BenchmarkValue::benchmark_value(),
+			destination_address: BenchmarkValue::benchmark_value(),
+			deposit_metadata: Some(BenchmarkValue::benchmark_value()),
+			swap_account: BenchmarkValue::benchmark_value(),
+			creation_slot: BenchmarkValue::benchmark_value(),
+			broker_fee: BenchmarkValue::benchmark_value(),
+			refund_params: BenchmarkValue::benchmark_value(),
+			dca_params: Some(BenchmarkValue::benchmark_value()),
+			boost_fee: BenchmarkValue::benchmark_value(),
+			affiliate_fees: BenchmarkValue::benchmark_value(),
+		}
+	}
+}
+
+pub struct SolanaVaultSwapsHandler;
+
+impl
+	SolanaVaultSwapAccountsHook<
+		VaultSwapAccountAndSender,
+		SolanaVaultSwapDetails,
+		SolanaTransactionBuildingError,
+	> for SolanaVaultSwapsHandler
+{
+	fn initiate_vault_swap(swap_details: SolanaVaultSwapDetails) {
+		SolanaIngressEgress::process_vault_swap_request(
+			swap_details.from,
+			swap_details.deposit_amount,
+			swap_details.to,
+			swap_details.destination_address,
+			swap_details.deposit_metadata,
+			Default::default(), // TODO txHash PRO-1760
```

Comment:
pro-1760 is done now, so I guess we can do this now?

Reply:
I'll do that afterwards in a separate PR on top of this or after it's merged to main. Already opened a ticket for that: https://linear.app/chainflip/issue/PRO-1790/solana-vault-swaps-update-transactioninid